pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -2151,6 +2152,18 @@ static void validateRequest(PinotQuery pinotQuery, int queryResponseLimit) {
         || !QueryOptionsUtils.isResponseFormatSQL(queryOptions)) {
       throw new IllegalStateException("SQL query should always have response format and group-by mode set to SQL");
     }
+    try {
+    // throw errors if options is less than 1 or invalid
+      Integer numReplicaGroupsToQuery = QueryOptionsUtils.getNumReplicaGroupsToQuery(queryOptions);
+      if (numReplicaGroupsToQuery != null) {
+        Preconditions.checkState(numReplicaGroupsToQuery > 0, "numReplicaGroups must be "
```

Comment:
If the user provided input is bad, it should result in a 4xx error. Preconditions check will result in a 5xx. Please fix it.

Reply:
PredConditions check will throw an IllegalStateException, which will be caught to set a response with a ProcessingException and error code QUERY_VALIDATION_ERROR_CODE = 700 [ BaseBrokerRequestHandler.java#L371](https://github.com/apache/pinot/blob/master/pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java#L371) In PinotClientRequest I believe it will just return this response

Reply:
> PredConditions check will throw an IllegalStateException, which will be caught to set a response with a ProcessingException and error code QUERY_VALIDATION_ERROR_CODE = 700 [ BaseBrokerRequestHandler.java#L371](https://github.com/apache/pinot/blob/master/pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java#L371) In PinotClientRequest I believe it will just return this response

Sorry for my incomplete understanding, and thanks for your clarification.